src/treegen.cpp

```diff
@@ -111,9 +111,23 @@ void make_tree(MMVManip &vmanip, v3s16 p0,
 }
 
 
+static void set_param_to_id(INodeDefManager *ndef, MapNode *n, u8 id)
+{
+	content_t t = n->getContent();
+	const ContentFeatures &f = ndef->get(t);
+	if (f.param_type_2 == CPT2_NONE) {
```

Comment:
Oh no!  If you're going to set a param to something, you NEED to define a content param type!  CPT_NONE implies there really is no data there.